### PR TITLE
Frontend/bkn 2933 adding share options in referral

### PR DESCRIPTION
**Pull request by aryanguptaSG**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
added share options in referral screen in driver app

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/android-native/app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -7,6 +7,7 @@
     android:id="@+id/main_layout"
     tools:context=".MainActivity">
     <include layout="@layout/splash" />
+     <include layout="@layout/referral_code" android:visibility="invisible"/>
```
Comment (Vignesh-772) [inline Frontend/android-native/app/src/main/res/layout/activity_main.xml:10]: no need to include layout  in `activity_main.xml. you can directly inflate the layout using inflater.

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

File: Frontend/android-native/app/src/main/res/drawable/referral_code_background.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="#FCC32C"/>
```
[the remark this answers is not in the document]
Comment (aryanguptaSG) [reply Frontend/android-native/app/src/main/res/drawable/referral_code_background.xml:3]: Done

File: Frontend/android-native/app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (aryanguptaSG) [reply Frontend/android-native/app/src/main/res/layout/activity_main.xml:10]: done

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 3]
```diff
@@ -933,6 +933,20 @@ public void run() {
...
                 }
                 break;
+            case 10:
+                SharedPreferences sharedPref = getSharedPreferences("application", Context.MODE_PRIVATE);
+                String path = sharedPref.getString("referral_image_path",null);
+                try {
+                    getContentResolver().delete(Uri.parse(path), null, null);
+                }
+                catch (Exception e){
+                    System.out.println("Aryan ->ex " +e);
```
[the remark this answers is not in the document]
Comment (aryanguptaSG) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:943]: done

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java [hunk 4]
```diff
@@ -2652,21 +2656,53 @@ public void run() {
...
+                        sendIntent.putExtra(Intent.EXTRA_TEXT, message);
+                        sendIntent.setType("*/*");
+                        sendIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                        Intent shareIntent = Intent.createChooser(sendIntent, null);
+
+                        SharedPreferences.Editor editor = sharedPref.edit();
+                        editor.putString("SHARE_INTENT_IS_OPEN", "true");
+                        editor.putString("referral_image_path",imagePath);
+                        editor.apply();
+                        activity.startActivityForResult(shareIntent , 10);
```
[the remark this answers is not in the document]
Comment (aryanguptaSG) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:2697]: done

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java [hunk 5]
```diff
@@ -80,6 +80,7 @@
 import android.view.SurfaceView;
 import android.view.View;
 import android.view.ViewGroup;
+import android.view.ViewTreeObserver;
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:83]: Why this import statement is required?

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java [hunk 6]
```diff
@@ -90,13 +91,15 @@
 import android.widget.ImageView;
 import android.widget.LinearLayout;
 import android.widget.NumberPicker;
+import android.widget.RelativeLayout;
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:94]: Not used

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java [hunk 7]
```diff
@@ -90,13 +91,15 @@
...
 import android.widget.TextView;
 import android.widget.TimePicker;
 import android.widget.Toast;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.annotation.RequiresApi;
+import androidx.constraintlayout.widget.ConstraintLayout;
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:102]: Import statement is not used

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java [hunk 8]
```diff
@@ -2656,21 +2660,51 @@ public void run() {
...
+                        Bitmap bitmap = Bitmap.createBitmap(imageView.getMeasuredWidth(), imageView.getMeasuredHeight(), Bitmap.Config.ARGB_8888);
+                        Canvas canvas = new Canvas(bitmap);
+                        imageView.draw(canvas);
+
+                        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+                        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, bytes);
+                        String imagePath = MediaStore.Images.Media.insertImage(context.getContentResolver(), bitmap, "IMG_" + Calendar.getInstance().getTime(), null);
+                        Uri uri = Uri.parse(imagePath);
+
+                        String message = "👋 Hey,\n\nMy Namma Yatri Referral Code is "+referralCode+".\n\nScan the QR code and download Namma Yatri app. You can help me out by entering my referral code on the Home screen.\n\nThanks!";
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:2687]: Why are we hardcoding `message` here? It should come from Jbridge

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java [hunk 9]
```diff
@@ -2656,21 +2660,51 @@ public void run() {
...
+                        editor.putString("SHARE_INTENT_IS_OPEN", "true");
+                        editor.putString("referral_image_path",imagePath);
+                        editor.apply();
+                        activity.startActivityForResult(shareIntent , REFERRAL_CODE_SHARE_ACTIVITY_RESULT_CODE);
+                    }
+                    else{
+                        ActivityCompat.requestPermissions(activity ,new  String []{READ_EXTERNAL_STORAGE , WRITE_EXTERNAL_STORAGE},1);
+                    }
+                }catch (Exception e){
+                    toast(e.getMessage());
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:2706]: Do not show toast for exceptions.

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java (hunk 5 above)
Comment (aryanguptaSG) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:83]: done

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java (hunk 6 above)
Comment (aryanguptaSG) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:94]: done

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java (hunk 7 above)
Comment (aryanguptaSG) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:102]: done

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java (hunk 9 above)
Comment (aryanguptaSG) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:2706]: done

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java (hunk 8 above)
Comment (aryanguptaSG) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:2687]: yes but we are using  referral code in layout so we can not pass whole msg from Jbridge , if we pass msg from Jbridge then we will have to use two arguments . that's why i am passing only code from Jbridge and using that in layout as well as in msg also .

**Comment by Vignesh-772** [comment]
squash the commits

**Comment by witcher-shailesh** [comment]
Will re-open if required

**Pull request closed without merging**